src/main/java/com/salesforce/storm/spout/sideline/handler/SidelineSpoutHandler.java

```diff
@@ -163,8 +187,8 @@ public void onSpoutOpen(
     /**
      * Loads existing sideline requests that have been previously persisted and checks to make sure that they are running.
      */
-    private void loadSidelines() {
-        final VirtualSpoutIdentifier fireHoseIdentifier = new DefaultVirtualSpoutIdentifier(getVirtualSpoutIdPrefix() + ":" + MAIN_ID);
+    void loadSidelines() {
```

Comment:
The logic in loadSidelines() seems suspect to me if we run it multiple times for the following reasons:

- I believe loadSidelines() should be marked synchronized since it can get called from multiple threads now, and we probably don't want concurrent executions of that method.
- We also need to be mindful of any variable loadSidelines() touches, and where else those variables are modified/read.  This could raise some complicated race conditions.
- It looks like loadSidelines() goes thru any "active" sidelines and attempts to re-add them, but then stops short with the hasVirtualSpout() call.  Since THAT call crosses some thread barriers we need to be careful about having consistent reads on what that's accessing.  Also it looks like that check is at the end, AFTER its modified the firehose's filter chain.  Meaning we could be adding things to the filter chain every 10 mins that a sideline is running.  We could have a weird race condition here between that being called and when a virtual spout is closing up.  Example:

Start VSpout A -> Adds FilterChain Step X to firehose
.. sometime passes
Running VSpout A -> close() -> triggers removal of the filterchain from firehose
Concurrently/interleaved run of loadSidelines() adds the filterchain back? Maybe even fires the sideline back up?

Reply:
I don't have an issue with making this synchronized, but it's not clear to me what else you are suggesting be done here?

Reply:
So I think I mis-read part of the logic in loadSidelines() but my concern around making it synchronized still makes sense.  

I also worry about odd race conditions between that method running and the state it checks against variables modified by other threads.  I wonder if it'd be better to notify the handler when a vspout instance terminates abnormally vs polling?

So one race condition I could imagine is if a vspout is closing (normally) and loadSidelines() runs at just the right moment, it may try to restart a closing vspout?

Reply:
It could restart the closing spout, potentially - but the new spout would be a no-op and close right after, so I don't think that's a huge concern honestly.  That closing spout is going to have a consumer id that matches the first and it's tracked state should match too.

All that said, I think you're onto something here... we should probably look at doing something more akin to an EventEmitter for a bunch of these signaling concerns. It would allow us to create hooks all over creation and solve some other concerns I have too.